### PR TITLE
adds to typeDefs.ts

### PR DESCRIPTION
**Pull request by mino323**
Next steps of https://github.com/garageScript/c0d3-app/issues/1192

This pr updates typedefs to include a way to create, read and delete modules. 

After writing I ran ```yarn generate``` to update index.tsx types.

Next step is to write the resolvers.

### CODE REVIEW

File: graphql/typeDefs.ts [hunk 1]
```diff
@@ -5,6 +5,7 @@ export default gql`
     lessons: [Lesson!]!
     session: Session!
     allUsers: [User]
+    allModules: [Module]
```
Comment (songz) [inline graphql/typeDefs.ts:8]: should this be `[Module]!` so we guarantee returning an array always?

Also, I think `modules` is better than `allModules`. `allUsers` should also be changed to `users` but that's for another time

File: graphql/typeDefs.ts [hunk 2]
```diff
@@ -53,6 +54,13 @@ export default gql`
       submissionId: Int!
       content: String!
     ): Comment
+    addModule(
+      authorId: Int!
+      lessonId: Int!
+      name: String!
+      content: String!
+    ): Module
+    deleteModule(id: Int, name: String!): SuccessResponse
```
Comment (songz) [inline graphql/typeDefs.ts:63]: I don't think name is needed.

Comment (mino323) [reply graphql/typeDefs.ts:63]: I'll try it out

**Review by flacial — APPROVED** [review]

File: graphql/typeDefs.ts [hunk 3]
```diff
@@ -206,4 +214,14 @@ export default gql`
     student: User!
     lesson: Lesson!
   }
+
+  type Module {
+    id: Int!
+    author: User!
+    authorId: Int!
```
Comment (songz) [inline graphql/typeDefs.ts:221]: authorId is inside the author object, so it is redundant here

File: graphql/typeDefs.ts [hunk 4]
```diff
@@ -206,4 +214,14 @@ export default gql`
...
+    lesson: Lesson!
+    lessonId: Int!
```
Comment (songz) [inline graphql/typeDefs.ts:223]: `lessonId` is inside the lesson object, so this is redundant here. Users could just use `lesson.id`

File: graphql/typeDefs.ts [hunk 5]
```diff
@@ -53,6 +54,13 @@ export default gql`
       submissionId: Int!
       content: String!
     ): Comment
+    addModule(
+      authorId: Int!
```
Comment (songz) [inline graphql/typeDefs.ts:58]: when you add module, `authorId` should not be in the input. This comes from the cookie. If you do it this way, it opens up a potential security flaw where I would be able to add a module and put someone else's id to pretend like someone else created the module.

**Pull request merged**